code-challenges.test.js

```diff
@@ -35,9 +64,38 @@ var fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
 var fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
 // Expected output: [-823, 7, 23]
 
+//a describe method to test that onlyOdds are returned in ascending order
+describe("onlyOdd", () => {
+  it("filters through the array and returns a new array of only odd numbers in ascending order", () => {
+    //example input: [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
+    //expected output : [-9, 7, 9, 199]
+    expect(onlyOdd(fullArr1)).toEqual([-9, 7, 9, 199]);
+    //example input: ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
+    //expected output: [-823, 7, 23]
+    expect(onlyOdd(fullArr2)).toEqual([-823, 7, 23]);
+  });
+})
+
+//ReferenceError: onlyOdd is not defined
 
 // b) Create the function that makes the test pass.
 
+//create a function that takes in an array
+const onlyOdd = (array) => {
+  let newArr = array
+  //parse through the array and return values that match type number
+  .filter((value) => typeof value === "number")
+  //calculates whether the number is odd
+  .filter((value) => {
+    //use ternary operator to return value if number is odd, else null
+    return value % 2 !== 0 ? value : null;
+  });
+    //use sort() method to return values in ascending order
+    return newArr.sort((index1, index2) => index1 - index2);
+};
+console.log(fullArr1);
+
```

Comment:
Looks good. I would suggest thinking about packing as much logic into a single filter method as you can. If you are evaluating for two things you can use a logical operator.